flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -272,11 +287,161 @@ private static ShuffleDescriptor createShuffleDescriptor(
 			ResultPartitionID resultPartitionID,
 			ResourceID location,
 			TaskManagerLocation senderLocation,
-			int channel) {
+			int connectionIndex) {
 		final NettyShuffleDescriptorBuilder builder = NettyShuffleDescriptorBuilder.newBuilder()
 			.setId(resultPartitionID)
 			.setProducerInfoFromTaskManagerLocation(senderLocation)
-			.setConnectionIndex(channel);
+			.setConnectionIndex(connectionIndex);
 		return localMode ? builder.setProducerLocation(location).buildLocal() : builder.buildRemote();
 	}
+
+	/**
+	 * A {@link SingleInputGateFactory} which replaces the default {@link RemoteInputChannel} and
```

Comment:
We should adjust the descriptions a bit. typo `costume`
E.g. A benchmark-specific input gate factory which overrides the respective methods of creating {@link RemotInputChannel} and {@link LocalInputChannel} for requesting specific subpartition.